google-cloud-bigquery/lib/google/cloud/bigquery/table.rb

```diff
@@ -1217,6 +1227,13 @@ def to_gapi
             @gapi
           end
 
+          def time_partitioning type: "DAY", expiration_ms: nil
```

Comment:
I'd say use milliseconds for now, and if there is a consensus that everything should be in seconds, then we'll have to discuss changing `Dataset#default_expiration` as well.